synapse/tests/test_tools_backup.py

```diff
@@ -0,0 +1,47 @@
+import os
+import synapse.lib.scope as s_scope
+
+import synapse.tests.utils as s_test
+
+import synapse.tools.backup as s_backup
+
+class BackupTest(s_test.SynTest):
+
+    def dirset(self, dir, skipfns):
+        ret = set()
+        for fdir, _, fns in os.walk(dir):
+            for fn in fns:
+                if fn in skipfns:
+                    continue
+                fp = os.path.join(fdir, fn)
+                fp = fp[len(dir):]
+                ret.add(fp)
+        return ret
+
+    def compare_dirs(self, dir1, dir2, skipfns=None):
+        if not skipfns:
+            skipfns = []
+        set1 = self.dirset(dir1, skipfns)
+        set2 = self.dirset(dir2, skipfns)
+        self.gt(len(set1), 1)
+        self.gt(len(set2), 1)
+        self.eq(set1, set2)
+        return set1
+
+    def test_backup(self):
+        with self.getTestCore() as core:
+            src_dirn = s_scope.get('dirn')
+            # This technically mangles the value in scope but that value
+            # is not used for doing directory removal.
+            with self.getTestDir() as dirn2:
+                argv = [src_dirn, dirn2]
+                args = s_backup.parse_args(argv)
+                ret = s_backup.main(args)
+                self.eq(ret, 0)
+                cmpr_path = os.path.join(args.outpath,
+                                         os.path.basename(src_dirn))
+                fpset = self.compare_dirs(src_dirn,
+                                         cmpr_path,
+                                         skipfns=['lock.mdb'])
+                # We expect the data.mdb file to be in the fpset
+                self.isin('/layers/default/layer.lmdb/data.mdb', fpset)
```

Comment:
Should the test make sure that the files are exactly the same, since no other operations should really occur during the execution of the unit test?

Reply:
Nope, my question is invalid